_rest_routes.inc.php

```diff
@@ -6234,7 +6237,247 @@
+     *          ref="#/components/responses/badrequest"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          ref="#/components/responses/unauthorized"
+     *      ),
+     *      security={{"openemr_auth":{}}}
+     *  )
+     */
+    "POST /api/patient/:pid/transaction" => function ($pid) {
+        RestConfig::authorization_check("patients", "trans");
+         $data = (array) (json_decode(file_get_contents("php://input")));
+         $return = (new TransactionRestController())->CreateTransaction($pid, $data);
+         RestConfig::apiLog($return, $data);
+         return $return;
+    },
 
+    /**
+     *  @OA\PUT(
+     *      path="/api/patient/transaction/{tid}",
+     *      description="Updates a transaction",
+     *      tags={"standard"},
+     *      @OA\Parameter(
+     *          name="tid",
+     *          in="path",
+     *          description="The id for the transaction.",
+     *          required=true,
+     *          @OA\Schema(
+     *              type="string"
+     *          )
+     *      ),
+     *      @OA\RequestBody(
+     *          required=true,
+     *          @OA\MediaType(
+     *              mediaType="application/json",
+     *              @OA\Schema(ref="#/components/schemas/api_transaction_request")
+     *          )
+     *      ),
+     *      @OA\Response(
+     *          response="200",
+     *          ref="#/components/responses/standard"
+     *      ),
+     *      @OA\Response(
+     *          response="400",
+     *          ref="#/components/responses/badrequest"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          ref="#/components/responses/unauthorized"
+     *      ),
+     *      security={{"openemr_auth":{}}}
+     *  )
+     */
+    "PUT /api/patient/transaction/:tid" => function ($tid) {
+        RestConfig::authorization_check("patients", "trans");
+         $data = (array) (json_decode(file_get_contents("php://input")));
+         $return = (new TransactionRestController())->UpdateTransaction($tid, $data);
+         RestConfig::apiLog($return, $data);
+         return $return;
+    },
+
+    /**
+     *  @OA\Get(
+     *      path="/api/user/transaction_type",
+     *      description="Get a list of transaction types",
+     *      tags={"standard"},
+     *      @OA\Response(
+     *          response="200",
+     *          ref="#/components/responses/standard"
+     *      ),
+     *      @OA\Response(
+     *          response="400",
+     *          ref="#/components/responses/badrequest"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          ref="#/components/responses/unauthorized"
+     *      ),
+     *      security={{"openemr_auth":{}}}
+     *  )
+     */
+
+    "GET /patient/transaction_type" => function () {
+        $return = (new TransactionRestController())->getTransactionTypes();
+        RestConfig::apiLog($return, $data);
```

Comment:
remove the $data parameter from this function (note no $data here)

Reply:
resolved